packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -19,37 +19,31 @@ typedef LayoutWidgetBuilder = Widget Function(BuildContext context, BoxConstrain
 
 /// An abstract superclass for widgets that defer their building until layout.
 ///
-/// Similar to the [Builder] widget except that the framework calls the [builder]
-/// function at layout time and provides the constraints that this widget should
-/// adhere to. This is useful when the parent constrains the child's size and layout,
-/// and doesn't depend on the child's intrinsic size.
+/// Similar to the [Builder] widget except that the implementation calls the [builder]
+/// function at layout time and provides the [LayoutInfo] that is required to
+/// configure the child widget subtree.
 ///
-/// {@template flutter.widgets.ConstrainedLayoutBuilder}
-/// The [builder] function is called in the following situations:
+/// This is useful when the child widget tree relies on information that are only
+/// available during layout, and doesn't depend on the child's intrinsic size.
 ///
-/// * The first time the widget is laid out.
-/// * When the parent widget passes different layout constraints.
-/// * When the parent widget updates this widget.
-/// * When the dependencies that the [builder] function subscribes to change.
+/// The equality of the [LayoutInfo] type is used by the implementation to avoid
+/// unnecessary rebuilds: if the new [LayoutInfo] computed during layout is the
+/// same as (defined by [LayoutInfo]'s [==] operator) the previous [LayoutInfo],
+/// the implementation will try to avoid calling the [builder] again unless
+/// [updateShouldRebuild] returns true.
```

Comment:
Could hint that maybe LayoutInfo types should be `@immutable` to avoid any food guns around this.